src/components/button/button.tsx

```diff
@@ -1,12 +1,33 @@
-import React, { InputHTMLAttributes } from 'react';
-import { StitcheButton } from './styles';
+import React from 'react';
+import { ButtonStyles } from './styles';
 
-export interface ButtonProps extends InputHTMLAttributes<HTMLButtonElement> {
+type StitchesButtonProps = React.ComponentProps<typeof ButtonStyles>;
+
+export type ButtonProps = {
   label: string;
-}
+  type?: string;
+  disabled?: boolean;
```

Comment:
```suggestion
  disabled?: boolean = false;
```

implicity this value